app/add_some_abilities/add_protection.py

```diff
@@ -0,0 +1,4 @@
+def protection(knight: dict) -> None:
+    knight["add_some_abilities"] = sum(
+        [part_of_armour["add_some_abilities"]
+         for part_of_armour in knight["armour"]])

```

Comment:
Consider creating a separate function to calculate the sum of 'add_some_abilities' from the knight's armour to adhere to the DRY principle. This will make the code more modular and reusable.